carp.protocols.core/src/commonMain/kotlin/dk/cachet/carp/protocols/domain/devices/BLEHeartRateSensor.kt

```diff
@@ -0,0 +1,41 @@
+package dk.cachet.carp.protocols.domain.devices
+
+import dk.cachet.carp.common.Trilean
+import dk.cachet.carp.common.data.DataType
+import dk.cachet.carp.protocols.domain.sampling.DataTypeSamplingSchemeList
+import dk.cachet.carp.protocols.domain.sampling.SamplingConfiguration
+import dk.cachet.carp.protocols.domain.sampling.carp.HeartRateSamplingScheme
+import dk.cachet.carp.protocols.domain.tasks.measures.DataTypeMeasure
+import kotlinx.serialization.Serializable
+import kotlin.reflect.KClass
+
+
+/**
+ * A Bluetooth Low Energy (BLE) device which implements a GATT Heart Rate service (https://www.bluetooth.com/specifications/gatt/services/).
+ */
+@Serializable
+data class BLEHeartRateSensor( override val roleName: String ) :
+    DeviceDescriptor<MACAddressDeviceRegistration, MACAddressDeviceRegistrationBuilder>()
+{
+    companion object
+    {
+        /**
+         * Measure the number of heart contractions (beats) per minute
+         */
+        fun heartRate() = DataTypeMeasure( SamplingSchemes.HEART_RATE.type )
```

Comment:
We should probably refactor `PhoneSensorMeasure` into `PassiveSensorMeasure` and use that here. Right now no `duration` can be specified. I think any measure options for internal phone sensors will be equally applicable here. E.g., later possible additions are until a certain data point count has been reached, or data capacity, or timeout in between data points, etc.